tsdb/head.go

```diff
@@ -805,6 +805,26 @@ func (h *Head) Init(minValidTime int64) error {
 	if err != nil && err != record.ErrNotFound {
 		return errors.Wrap(err, "find last checkpoint")
 	}
+
+	// Find the last segment.
+	// The purpose of this scope is to avoid wal.Segments err override wal.LastCheckpoint err.
+	var endAt int
+	{
+		var err error
+		_, endAt, err = wal.Segments(h.wal.Dir())
+		if err != nil {
+			return errors.Wrap(err, "finding WAL segments")
+		}
+	}
```

Comment:
Not so comfortable having `err` in between when we are consuming some other `err` below this. How about this?
```suggestion
	_, endAt, e := wal.Segments(h.wal.Dir())
	if e != nil {
		return errors.Wrap(e, "finding WAL segments")
	}
```